lib/interactive_logger/step.rb

```diff
@@ -60,9 +61,20 @@ def prefix(str)
     end
 
     def print_msg(str)
+      str = terminal_trim(str)
       blank
       @last_print_msg = str
       print str
     end
+
+    # Trim string to current terminal width and break at newline so it can be
+    # replaced rather than causing the same line to print multiple times.
+    def terminal_trim(str)
+      terminal_width = @thor_shell.terminal_width
+      if str.length > (terminal_width - 18)
+        str = "#{str[0..(terminal_width - 22)]}..."
+      end
+      str.lines.first.chomp
```

Comment:
This should happen before the string is trimmed, otherwise you might get a truncated string without ellipsis.